src/agenteval/cli.py

```diff
@@ -563,5 +567,115 @@ def eval_command(
+    required=True,
+    help="HF file with results to rescore, e.g. 1.0.0-dev1/validation/miked-ai_ReAct_reasoning_claude-sonnet-4-20250514_2025-07-10T21-46-47.json",
+)
+@click.option(
+    "--resubmission-prefix",
+    type=str,
+    required=True,
+    help="Path prefix where the resubmitted logfiles and results will be uploaded, e.g. 1.0.0-rescored/validation",
+ )
+@click.option(
+    "--work-dir",
+    type=str,
+    default=None,
+    help="Working directory for downloading and processing results. If missing, use a temporary directory.",
+)
+def rescore_command(
+    results_repo_id: str,
+    submissions_repo_id: str,
+    results_path: str,
+    resubmission_prefix: str,
+    work_dir: str | None = None,
+):
+    """Download existing results from a HF dataset, recompute summaries, and upload to a new dataset."""
+    temp_dir = None
+    if work_dir is None:
+        temp_dir = tempfile.TemporaryDirectory()
+        work_dir = temp_dir.__enter__()
+
+    local_results_dir = os.path.join(work_dir, "results")
+    local_submissions_dir = os.path.join(work_dir, "submissions")
+    from huggingface_hub import HfApi
+
+    hf_api = HfApi()
+
+    click.echo(f"Downloading results from {results_repo_id}/{results_path}")
+    hf_api.hf_hub_download(
+        repo_id=results_repo_id,
+        repo_type="dataset",
+        filename=results_path,
+        local_dir=local_results_dir,
+    )
+    local_results_path = os.path.join(local_results_dir, results_path)
+    with open(local_results_path) as f:
+        eval_result = EvalResult.model_validate(json.load(f))
+
+
+    submissions_path = results_path.replace(".json", "")
+    local_agenteval_path = os.path.join(local_submissions_dir, submissions_path, "agenteval.json")
+    click.echo(f"Downloaded raw logs from {submissions_repo_id}/{submissions_path}")
+    hf_api.snapshot_download(
+        repo_id=submissions_repo_id,
+        repo_type="dataset",
+        allow_patterns=f"{submissions_path}/*",
+        local_dir=local_submissions_dir,
+    )
+    click.echo(f"Recomputing results. Overwriting existing results in {local_agenteval_path}.")
+    task_results, eval_specs, had_errors = process_eval_logs(
+        os.path.join(local_submissions_dir, submissions_path)
+    )
+    eval_result.eval_specs = eval_specs
+    eval_result.results = task_results
+    eval_result.save_json(local_results_path)
+
+    submission_name = submissions_path.split("/")[-1]
+    resubmission_path = f"{resubmission_prefix}/{submission_name}"
+    click.echo(f"Uploading submission to {submissions_repo_id}/{resubmission_path}")
+    hf_api.upload_folder(
+        repo_id=submissions_repo_id,
+        repo_type="dataset",
+        folder_path=os.path.join(local_submissions_dir, submissions_path),
+        path_in_repo=resubmission_path,
+    )
+
+    compressed_usages_result = compress_model_usages(eval_result)
+    click.echo(f"Uploading results to {results_repo_id}/{resubmission_path}.json")
+    hf_api.upload_file(
```

Comment:
Working towards verifying this, but I'm thinking if we're sending to a potentially new resubmission prefix, we'll want to create/update the hf schema as we do [here](https://github.com/allenai/agent-eval/blob/main/src/agenteval/leaderboard/upload.py#L175)

Would it make sense to use the existing `upload_summary_to_hf`? It seems possible that we may want to add additional validation logic over time, so having all paths use the same upload method could make sure that all paths (cli publish, cli rescore, and ui submit) have the same sanitization/validation